TransformationPlayground/playground/transform_expert/expert.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+from dataclasses import dataclass
+import logging
+from typing import Any, Callable, Dict, List
+
+from botocore.config import Config
+from langchain_aws import ChatBedrockConverse
+from langchain_core.language_models import LanguageModelInput
+from langchain_core.messages import BaseMessage, SystemMessage, ToolMessage
+from langchain_core.runnables import Runnable
+
+from transform_expert.parameters import SourceVersion, TargetVersion, TransformType, TransformLanguage
+from transform_expert.prompting import get_system_prompt_factory
+from transform_expert.tools import ToolBundle, get_tool_bundle
+from transform_expert.utils.transforms import TransformTask
+
+
+logger = logging.getLogger("transform_expert")
+
+
+@dataclass
+class Expert:
+    llm: Runnable[LanguageModelInput, BaseMessage]
+    system_prompt_factory: Callable[[Dict[str, Any]], SystemMessage]
+    tools: ToolBundle
+
+def get_expert(source_version: SourceVersion, target_version: TargetVersion, transform_type: TransformType, transform_language: TransformLanguage) -> Expert:
+    logger.info(f"Building expert for: {source_version}, {target_version}, {transform_type}, {transform_language}")
+
+    # Get the tool bundle for the given transform language
+    tool_bundle = get_tool_bundle(transform_language)
+
+    # Define a boto Config to use w/ our LLM that's more resilient to long waits and frequent throttling
+    config = Config(
+        read_timeout=120, # Wait 2 minutes for a response from the LLM
+        retries={
+            'max_attempts': 20,  # Increase the number of retry attempts
+            'mode': 'adaptive'   # Use adaptive retry strategy for better throttling handling
+        }
+    )
+
+    # Define our Bedrock LLM and attach the tools to it
+    llm = ChatBedrockConverse(
+        model="anthropic.claude-3-5-sonnet-20240620-v1:0", # This is the older version of the model, could be updated
```

Comment:
Claude 3.5 Sonnet (v1) is industry-leading at general coding tasks of this sort.  In fact, I think it's probably \*too\* good at those tasks and may be overkill.  I'd like to swap to the new Claude Haiku model when it drops and see if we can get away w/ a smaller, faster, cheaper model here.  I chose Sonnet because it's a sledgehammer that I have high confidence will "drive the nail".  In the longer term, I'd like to have a model where the first pass of each transformation process is done w/ a small, fast, cheap model and we switch to a more capable model if we encounter issues in our GenAI test-driven development loop.

Regarding effort to swap models - I expect them to be highly interchangeable with the exception that our prompt template is tied to the model and version.  Anthropic models, like Claude, prefer to receive in-prompt directives via XML tags, where Meta models, like Llama, have their own syntax (I think they use `$`'s or something).  So swapping a model would entail updating this text here along with possibly re-writing and re-testing our system prompt.  We'll ideally want some benchmark tests we can run against prospective models/prompts to help us compare/onboard.